### DISCUSSION TITLE
grid_dim.y boundary check

### DISCUSSION THREAD

**Discussion by zhaolianshuizls** [NVIDIA/cccl#1059] [CUB] [answered]
Is it possible `num_tiles` is so large that it exceeds the `grid_dim.y` boundary?
https://github.com/NVIDIA/cub/blob/b2e8bccb8c0cd15279974fe4b9b8d6fcd1842b57/cub/device/dispatch/dispatch_select_if.cuh#L422

**Comment by gevtushenko** [comment, marked as answer]
Hello, @zhaolianshuizls! The code you mentioned is written in terms of tiles. Tile size depends on the data type, but let's say it's 128 threads, each processing ten items, so the tile size is 1280 (~2^10) items. To calculate the number of tiles, select first checks the maximum x-dimension of a grid of thread blocks. It should be around 2^31 blocks. This already converts to 2^41 items without increasing the number of blocks along the Y dimension. If you are using our device API, the `num_items` type is currently limited to be an `int`, so it should be impossible to get `scan_grid_size.y > 1`. If you are using the dispatch layer directly, providing `OffsetT` different from `int`, it should be possible to get `scan_grid_size.y > 1`. In this case, the maximal grid size along the y dimension is 2^16, giving us the `num_items` limit of about 2^57 items.
